content/docs/contributing/contributing_changes.md

```diff
@@ -29,133 +27,148 @@ In short
 
 This workflow should help changes be deployed quickly.
 
+## Creating a Branch
+
+You should make your changes on a new branch, and not on `main`.
+This will make it easier to merge changes from many contributors.
+Try to pick a branch name which is short and describes the change you are making.
+
+{{% hint warning %}}
+The changes on one branch or pull request should address a
+single issue and be self-contained.
+Don't try to solve more than one unrelated problem at once.
+
+Sticking to this practice will help ensure pull requests are small and easy to review.
+Otherwise, you might end up having to make many changes during review, have difficultly keeping your branch up to date or upset the reviewers.
+{{% /hint %}}
+
+## Making Changes
+
+At this point, you can edit files and create `git` commits as necessary.
+
 {{% hint info %}}
 When making commits, it is a good idea to follow Tim Pope's
 [recommendations](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html)
 for good commit messages.
 {{% /hint %}}
 
-## Creating a Branch
+It is often useful to run the Markdown style checker (and other tests) before committing.
+This allows you to catch, and fix, potential issues before opening a pull request.
+To do so, use the `pre-commit` tool.
+From the repository root, run:
+
+```bash
+brew install pre-commit  # if not already installed
+pre-commit install
+```
 
-You should make your changes on a new branch, and not on `main`. This will make
-it easier to merge changes from many contributors. Try to pick a branch name
-which is short and describes the change you are making.
+This creates a pre-commit [Git hook](https://git-scm.com/book/en/v2/Customizing-Git-Git-Hooks),
+which (as the name suggests) will be automatically run every time before a commit is created.
+It currently performs the following checks, which are designed to mimic those in [Continuous Integration](#continuous-integration):
 
-{{% hint warning %}} The changes on one branch or pull request should address a
-single issue and be self-contained. Don't try to solve more than one unrelated
-problem at once.
+- Markdown linting, i.e. style checking:
+- a Hugo build, to make sure that the pages can be built; and
+- a Markdown link check.
 
-Sticking to this practice will help ensure pull requests are small and easy to
-review. Otherwise, you might end up having to make many changes during review,
-have difficultly keeping your branch up to date or upset the reviewers.  {{%
-/hint %}}
+If you want to manually perform these checks at any point in time, use:
+
+```bash
+pre-commit run
```

Comment:
In practice, when I tried following these steps I ended up with errors, presumably because I only had the default Ruby installed:
```
An unexpected error has occurred: CalledProcessError: command: ('/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby', '/usr/bin/gem', 'install', '--no-document', '--no-format-executable', '--no-user-install', 'mdl-0.12.0.gem')
return code: 1
stdout: (none)
stderr:
    ERROR:  Error installing mdl-0.12.0.gem:
        There are no versions of mdl (= 0.12.0) compatible with your Ruby & RubyGems
        mdl requires Ruby version >= 2.7. The current ruby version is 2.6.10.210.
Check the log at /Users/dllewellyn-jones/.cache/pre-commit/pre-commit.log```
```
To get it to work I had to do the following (probably there's a nicer way):
```
brew install ruby
PATH="$(brw --prefix ruby)/bin:$PATH" pre-commit run
```
Is there any way to capture this to avoid tripping people up, without making the instructions overly complex?

Reply:
Hmmm, that's not good. I was under the impression that `pre-commit` would reinstall its own copy of ruby and mdl. (The docs certainly seemed to suggest that [it installs its own copy of Node](https://pre-commit.com/#usage), for example, so I assumed the same was true for other languages.)

I'll investigate more (probably after Away Day)...

Thanks for giving it a go! It definitely needed it! I'll mark this as draft in the meantime I think.

Reply:
(I think you can do `brew link ruby --force --overwrite` which should create an appropriate symlink, but `brew` itself does suggest that editing `$PATH` is preferable to that, so I'm not sure how much of a good idea that is.)